### PR TITLE
Double to string for #44

### PR DESCRIPTION
**Pull request by ammaratef45**
PR for #44 
Changed double to string when referring to balance.

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#45](https://github.com/amihaiemil/zold-java-client/issues/45) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#45](https://github.com/amihaiemil/zold-java-client/issues/45) is assigned to @g4s8[/z](https://www.0crat.com/u/g4s8), here is [why](https://www.0crat.com/footprint/GFCHY7NQG/03cff25d-63be-4c54-b063-7455e96439d9); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @amihaiemil[/z](https://www.0crat.com/u/amihaiemil) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be no monetary reward for this job

File: src/main/java/com/amihaiemil/zold/RtWallet.java [hunk 1]
```diff
@@ -143,10 +144,27 @@ public void pay(
     }
 
     @Override
-    public void find(final String id, final String details) {
-        throw new UnsupportedOperationException(
-            "Not yet implemented. If you can contribute please, do it at "
-            + "https://github.com/amihaiemil/zold-java-client"
+    public JsonArray find(
```
Comment (g4s8) [inline src/main/java/com/amihaiemil/zold/RtWallet.java]: @ammaratef45 these changes are not related to origin issue #44, most probably there's another ticket to implement it, if you want to implement this method you can ask ARC to assign you that ticket and submit new PR with these changes.

File: src/main/java/com/amihaiemil/zold/RtWallet.java [hunk 2]
```diff
@@ -113,7 +114,7 @@ public double balance() throws IOException{
     @Override
     public void pay(
         final String keygap, final String user,
-        final double amount, final String details
+        final String amount, final String details
```
Comment (g4s8) [inline src/main/java/com/amihaiemil/zold/RtWallet.java]: @ammaratef45 this change will affect JSON payload format, type `amount` changed from `JSON-number` to `JSON-string`:
before:
```json
{
  "amount": 1234.56
}
```
after:
```json
{
  "amount": "1234.56"
}
``` 

I'm not sure here (because I didn't find API docs for JSON payload), but most probably API won't accept such format.

File: src/main/java/com/amihaiemil/zold/Wallet.java [hunk 3]
```diff
@@ -60,14 +62,17 @@
...
-    */
-    void find(String id, String details)throws IOException;
+     * Finds all payments that match this query and returns.
+     * @return Array of found transactions.
+     * @param id Wallet id
+     * @param details Regex of payment details
+     * @throws IOException If there's a networking problem.
+     * @throws URISyntaxException If there's a problem building URI.
+     */
+    JsonArray find(String id, String details)
```
Comment (g4s8) [inline src/main/java/com/amihaiemil/zold/Wallet.java]: @ammaratef45 same here - it's related to another ticket

File: src/main/java/com/amihaiemil/zold/RtWallet.java (hunk 1 above)
Comment (ammaratef45) [reply src/main/java/com/amihaiemil/zold/RtWallet.java]: @g4s8 They are in a different PR, I made those changes on the master branch of my fork and had to get them with me as the other PR is not closed

File: src/main/java/com/amihaiemil/zold/RtWallet.java (hunk 2 above)
Comment (ammaratef45) [reply src/main/java/com/amihaiemil/zold/RtWallet.java]: @g4s8 reverting it makes the issue meaningless, what do you think? we may need to rediscuss the issue, right?

Comment (g4s8) [reply src/main/java/com/amihaiemil/zold/RtWallet.java]: @ammaratef45 you can use `json.add("amount", new BigDecimal(amount))` - that code adds amount as json number, but with arbitrary precision

Comment (ammaratef45) [reply src/main/java/com/amihaiemil/zold/RtWallet.java]: @g4s8 The problem is that this change meant to be used for avoiding any casting or precision problems and let the higher layers handle this.

File: src/main/java/com/amihaiemil/zold/RtWallet.java (hunk 1 above)
Comment (g4s8) [reply src/main/java/com/amihaiemil/zold/RtWallet.java]: @ammaratef45 it looks like this PR doesn't depend on #43, so you can just submit two independent PRs

File: src/main/java/com/amihaiemil/zold/RtWallet.java (hunk 2 above)
Comment (g4s8) [reply src/main/java/com/amihaiemil/zold/RtWallet.java]: @ammaratef45 but `BigDecimal` doesn't limit the precision, so I think it's safe to use here. In any case we should conform to wts API protocol, so if you have any doubts you can submit a ticket to wts repo, e.g. to expand documentation for JSON apis.
// cc @amihaiemil

Comment (denis-konovalyenko) [reply src/main/java/com/amihaiemil/zold/RtWallet.java]: @ammaratef45 , @g4s8 , @amihaiemil , the following conversion is performed in `wts.rb` [here](https://github.com/zold-io/wts.zold.io/blob/7e3507fe96cd8cc6a0bd32a22ac1ab17644cc3e1/wts.rb#L464):
```
  amount = Zold::Amount.new(zld: params[:amount].to_f)
```
I am not a Ruby expert but my guess, which is based on [the documentation](https://apidock.com/ruby/String/to_f):
```
"123.45e1".to_f        #=> 1234.5
"45.67 degrees".to_f   #=> 45.67
"thx1138".to_f         #=> 0.0
```
is that
```
"amount": "1234.56"
```
would be interpretted as `1234.56` (Float).

Hope, that would help us out here. By the way, there has been opened a related to accurate decimal operations issue [#718](https://github.com/zold-io/zold/issues/718) in Zold project.

**Comment by ammaratef45** [comment]
@g4s8 I've merged the master and the irrelevant changes went away
Also, according to @denis-konovalyenko comment, looks like the string payload of amount won't make a problem (or we need to wait for zold-io/zold#718 )
What do you think?

**Comment by g4s8** [comment]
@ammaratef45 thanks

**Comment by g4s8** [comment]
@rultor merge

**Comment by amihaiemil** [comment]
@rultor merge it

**Comment by 0crat** [comment]
Job was finished in 24 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
The job [#45](https://github.com/amihaiemil/zold-java-client/issues/45) is now out of scope

**Comment by 0crat** [comment]
Order was finished: +20 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @amihaiemil[/z](https://www.0crat.com/u/amihaiemil)
